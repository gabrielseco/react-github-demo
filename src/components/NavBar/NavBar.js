// @flow
import React from 'react';
import styles from './NavBar.scss';

const NavBar = ({ children }) => (
  <nav className={styles.navBar}>
    <ul className={styles.navWrapper}>{children}</ul>
  </nav>
);

export default NavBar;
