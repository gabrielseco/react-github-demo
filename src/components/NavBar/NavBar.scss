.navBar {
  display: flex;
  flex: 2;
  padding: 0 20px;
}

.navWrapper {
  display: inline-block;
}
