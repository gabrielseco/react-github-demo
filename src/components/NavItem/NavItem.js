// @flow
import React from 'react';
import { Link } from 'react-router-dom';
import styles from './NavItem.scss';

const NavItem = ({ URL, children }) => (
  <li className={styles.navItem}>
    <Link className={styles.navLink} to={URL}>
      {children}
    </Link>
  </li>
);

export default NavItem;
