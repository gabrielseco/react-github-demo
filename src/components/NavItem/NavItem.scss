$nav-item-link-color: #212240;

.navItem {
  display: inline-block;
  list-style-type: none;
  margin: 0 20px;
}

.navLink {
  color: $nav-item-link-color;
}
