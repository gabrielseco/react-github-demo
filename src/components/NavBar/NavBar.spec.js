import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import NavBar from './NavBar';
import { NavItem } from './../NavItem';

describe('NavBar suite', () => {
  it('should render the NavBar component', () => {
    const navigation = (
      <NavBar>
        <NavItem URL="/repos">Repos</NavItem>
        <NavItem URL="/orgs">orgs</NavItem>
      </NavBar>
    );
    const component = shallow(navigation);
    const tree = renderer.create(<Router>{navigation}</Router>).toJSON();
    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();

    expect(component.find(NavItem).length).toBe(2);
  });
});
