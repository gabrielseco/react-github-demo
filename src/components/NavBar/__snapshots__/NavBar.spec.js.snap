// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar suite should render the NavBar component 1`] = `
<nav
  className="navBar"
>
  <ul
    className="navWrapper"
  >
    <li
      className="navItem"
    >
      <a
        className="navLink"
        href="/repos"
        onClick={[Function]}
      >
        Repos
      </a>
    </li>
    <li
      className="navItem"
    >
      <a
        className="navLink"
        href="/orgs"
        onClick={[Function]}
      >
        orgs
      </a>
    </li>
  </ul>
</nav>
`;
